src/newViews/UserGroup/AddUsergroupModal/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useCallback, useState } from 'react';
+import { isDefined } from '@togglecorp/fujs';
+import {
+    Modal,
+    Button,
+    PendingMessage,
+    TextInput,
+} from '@the-deep/deep-ui';
+import {
+    ObjectSchema,
+    PartialForm,
+    requiredCondition,
+    useForm,
+} from '@togglecorp/toggle-form';
+
+import { useLazyRequest } from '#utils/request';
+import NonFieldError from '#components/ui/NonFieldError';
+import _ts from '#ts';
+import styles from './styles.scss';
+
+interface UsergroupAdd {
+    title: string;
+    description?: string;
+}
+
+type FormType = {
+    id?: number;
+    title: string;
+};
+
+export interface Membership {
+    id: number;
+    member: number;
+    memberName: string;
+    memberEmail: string;
+    role: string;
+    group: number;
+    joinedAt: string;
+}
+
+export interface Usergroup {
+    id: number;
+    title: string;
+    description: string;
+    role: string;
+    memberships: Membership[];
+    globalCrisisMonitoring: boolean;
+    createdAt: string;
+    modifiedAt: string;
+}
+
+type FormSchema = ObjectSchema<PartialForm<FormType>>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        title: [requiredCondition],
+    }),
+};
+
+const defaultFormValue: PartialForm<FormType> = {};
+
+
+interface Props {
+    onModalClose: () => void;
+    onSuccess: () => void;
+    value?: Usergroup,
```

Comment:
We the pull latest data for a edit modal from inside the modal.
The advantages of this approach are:
1. The user always edits on the latest data
2. The modal will be re-usable. The user doesn't need to find all the data needed to edit the modal.
There might be certain cases when we may have only part of the entity we want to edit, in that case the form will have some missing value.

What do you think @AdityaKhatri @frozenhelium